controlplane/kubeadm/internal/etcd_client_generator.go

```diff
@@ -60,28 +57,30 @@ func (c *etcdClientGenerator) forNode(ctx context.Context, name string) (*etcd.C
 	return customClient, nil
 }
 
-// forLeader takes a list of nodes and returns a client to the leader node
-func (c *etcdClientGenerator) forLeader(ctx context.Context, nodes *corev1.NodeList) (*etcd.Client, error) {
-	var errs []error
-
-	for _, node := range nodes.Items {
-		client, err := c.forNode(ctx, node.Name)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-		defer client.Close()
-		members, err := client.Members(ctx)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
-		for _, member := range members {
-			if member.ID == client.LeaderID {
-				return c.forNode(ctx, member.Name)
-			}
-		}
+func (c *etcdClientGenerator) forAny(ctx context.Context, nodes *corev1.NodeList) (*etcd.Client, error) {
+	endpoints := make([]string, nodes.Size())
+	for i, node := range nodes.Items {
+		endpoints[i] = staticPodName("etcd", node.Name)
 	}
 
-	return nil, errors.Wrap(kerrors.NewAggregate(errs), "could not establish a connection to the etcd leader")
+	p := proxy.Proxy{
+		Kind:       "pods",
+		Namespace:  metav1.NamespaceSystem, // TODO, can etcd ever run in a different namespace?
```

Comment:
can get rid of `// TODO, can etcd ever run in a different namespace?`

it's never not going be `kube-system` for kubeadm.